File: 프로그래머스/Lv2/멀리 뛰기.py
def solution(n):
    answer = 0
    arr=[1,2]
    
    for i in range(2,n):
        arr.append((arr[i-1]%1234567+arr[i-2]%1234567)%1234567)
    
    answer=arr[n-1]
    return answer

File: 프로그래머스/Lv2/멀리 뛰기.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

long long solution(int n) {
    long long answer = 1;
    long long DP[2001];
    fill(DP, DP+2001, 0);
    
    DP[1]=1;
    DP[2]=2;
    
    for(int i=3;i<=n;i++){
        DP[i]=(DP[i-1]%1234567+DP[i-2]%1234567)%1234567;
    }
    
    answer=DP[n];
    
    return answer;
}
